registry/server/tag_cleanup.go

```diff
@@ -0,0 +1,116 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var TagMarkTimeKey = "tagMarkedForDeletionTime"
+var topicRegex = regexp.MustCompile(".*")
+
+type TagCleaner struct {
+	running bool
+}
+
+// RunTagCleanup is regularly checks for tags that are stale and need clean up.
+func (tc *TagCleaner) RunTagCleanup(s *Server, ctx context.Context, c Config) {
+	tc.running = true
+
+	// Interval timer.
+	t := time.NewTicker(time.Duration(c.TagCleanupFrequencyMinutes) * time.Second)
+	defer t.Stop()
+
+	for tc.running {
+		<-t.C
+		s.MarkForDeletion(time.Now)
```

Comment:
This returns an error; do we just optimistically progress to DeleteStaleTags anyway? From my brief skimming, that seems ok to do.